code/modules/mob/living/carbon/brain/brain_item.dm

```diff
@@ -69,23 +68,36 @@
 	owner.brain_op_stage = 4.0
 
 	var/obj/item/organ/brain/B = src
-	if(istype(B) && istype(owner))
+	if(istype(B) && istype(owner) && owner.internal_organs_by_name[organ_tag] == src)

```

Comment:
Yeah, this is highly problematic--you've always been able to tell who's brain is who's--and that *should* be something you can do.